proto/cosmos/base/reflection/v1beta1/reflection.proto

```diff
@@ -3,10 +3,161 @@ package cosmos.base.reflection.v1beta1;
+  string fullname = 1;
+  // interface_accepting_messages contains information regarding the proto messages which contain the interface as google.protobuf.Any field
+  repeated InterfaceAcceptingMessageDescriptor interface_accepting_messages = 2;
+  // interface_implementers is a list of the descriptors of the interface implementers
+  repeated InterfaceImplementerDescriptor interface_implementers = 3;
+}
+
+// InterfaceImplementerDescriptor describes an interface implementer
+message InterfaceImplementerDescriptor {
+  // fullname is the protobuf queryable name of the interface implementer
+  string fullname = 1;
+  // type_url defines the type URL used when marshalling the type as any
+  // this is required so we can provide type safe google.protobuf.Any marshalling and
+  // unmarshalling, making sure that we don't accept just 'any' type
+  // in our interface fields
+  string type_url = 2;
+}
+
+// InterfaceAcceptingMessageDescriptor describes a protobuf message which contains
+// an interface represented as a google.protobuf.Any
+message InterfaceAcceptingMessageDescriptor {
+  // fullname is the protobuf fullname of the type containing the interface
+  string fullname = 1;
+  // field_descriptor_names is a list of the protobuf name (not fullname) of the field
+  // which contains the interface as google.protobuf.Any (the interface is the same, but
+  // it can be in multiple fields of the same proto message)
+  repeated string field_descriptor_names = 2;
+}
+
+// ConfigurationDescriptor contains metadata information on the sdk.Config
+message ConfigurationDescriptor {
+  // bech32_account_address_prefix is the account address prefix
+  string bech32_account_address_prefix = 1;
+}
+
+// MsgDescriptor describes a cosmos-sdk message that can be delivered with a transaction
+message MsgDescriptor {
+  // msg contains a descriptor of sdk.ServiceMsg, note: sdk.Msg is not supported
+  // as every sdk.Msg is already an sdk.ServiceMsg. It is defined as a oneof in case
+  // different representation of a msg will be implemented.
+  oneof msg {
+    // service_msg is used when the message is an sdk.ServiceMsg type
+    ServiceMsgDescriptor service_msg = 1;
+  }
+}
+
+// ServiceMsgDescriptor describes an sdk.ServiceMsg type
+message ServiceMsgDescriptor {
+  // request_fullname is the protobuf fullname of the given sdk.ServiceMsg request
+  // this is the protobuf message type which should be used as google.protobuf.Any.value
+  // when delivering the msg to the DeliverTx endpoint
+  string request_fullname = 1;
+  // request_route is the sdk.ServiceMsg route, it is equal to type_url
+  string request_route = 2;
+  // request_type_url is the identifier that should be used as google.protobuf.Any.type_url
+  // when delivering the msg to the DeliverTx endpoint
+  string request_type_url = 3;
+  // response_fullname is the protobuf fullname of the given sdk.ServiceMsg response
+  string response_fullname = 4;
+}
```

Comment:
IMO you just need one field here: string fullname of the service method. Then, you can use a separate reflection (e.g. the gogoreflection one) to query the request_fullname and the response_fullname of each method

Reply:
The configurator is not inside the BaseApp, hence we have no access to MsgServer descriptors. This is documented in the implementation.